client/js/components/user/DpUserList/DpUserFormFields.vue

```diff
@@ -348,14 +349,17 @@ export default {
     getOrgaAllowedRoles (orgaId) {
       let allowedRoles = this.rolesInRelationshipFormat
 
-      this.fetchOrgaById(orgaId).then((orga) => {
-        this.setOrga(orga.data.data)
-        if (this.currentUserOrga.id && hasOwnProp(this.organisations[this.currentUserOrga.id].relationships, 'allowedRoles')) {
-          allowedRoles = this.organisations[this.currentUserOrga.id].relationships.allowedRoles.list()
-        }
-      })
+      return this.fetchOrgaById(orgaId)
+        .then(orga => {
+          this.setOrga(orga.data.data)
 
-      return allowedRoles
+          if (this.currentUserOrga.id && hasOwnProp(this.organisations[this.currentUserOrga.id].relationships, 'allowedRoles')) {
+            allowedRoles = this.organisations[this.currentUserOrga.id].relationships.allowedRoles.list()
+          }
+        })
+        .finally(() => {
+          return allowedRoles
+        })
```

Comment:
Do we always want to return `allowedRoles` even if the then block does not execute?

Reply:
I'm not sure. Does it make sense to return roles in line 350 if the request fails? I think it could be maybe confusing if the user does not see all the roles he is expecting. On the other hand, is it better if we return an empty array in case of error and the user cannot select any roles? I think you can leave it like this.

Reply:
returning an empty array would lead to an not savable user since role is an required field